Silence libdc1394 error

drothlis opened the pull request:
Thanks to Dileep Pothanchery (@dileep-p) for the suggestion.

Fixes https://github.com/stb-tester/stbt-docker/issues/5

Before this fix:

```
$ ./stbt-docker python
>>> import stbt
libdc1394 error: Failed to initialise libdc1394
```

drothlis committed:
Silence libdc1394 error

Thanks to Dileep Pothanchery (@dileep-p) for the suggestion.

Fixes https://github.com/stb-tester/stbt-docker/issues/5

Before this fix:

    $ ./stbt-docker python
    >>> import stbt
    libdc1394 error: Failed to initialise libdc1394

drothlis committed:
tests: Add test_stbt_match

We didn't have any tests that you could actually run `stbt` inside
stbt-docker! This gives me some confidence that the change in the
previous commit doesn't break anything.

wmanley commented on stbt-docker:
```diff
@@ -382,6 +382,7 @@ def main(argv):
               -o -print0 | xargs -0 chown -f {uid}:{gid} &&
          sed -i s/stb-tester:x:1000:1000/stb-tester:x:{uid}:{gid}/ /etc/passwd &&
          sed -i s/stb-tester:x:1000/stb-tester:x:{gid}/ /etc/group &&
+         ln -s /dev/null /dev/raw1394 &&
```
This does make me a little nervous.  I'm thinking about the case where you have `/dev` mounted in from the host.  Maybe it's fine.  Also I don't think we would want `stbt-docker` to fail if `/dev/raw1394` existed or `/dev` was for some reason unwritable.

drothlis replied:
> `/dev` mounted in from the host. 

Good point. I'll have a think about what could be done to address that.

drothlis committed:
Don't create dummy /dev/raw1394 if /dev mounted from host

We create `/dev/raw1394` (a symlink pointing at `/dev/null`) to silence
an error every time a Python script imports stbt. However we don't want
to create this symlink on the host if `/dev` is mounted in from the
host!

In my experiments (with both docker 1.6.2 and 1.12.3) `/dev` is devtmpfs
if you're mounting in `/dev` from the host, otherwise it's tmpfs:

    $ docker run --rm -ti ubuntu:14.04 cat /proc/mounts | grep devtmpfs

    $ docker run --rm -ti -v /dev:/dev ubuntu:14.04 cat /proc/mounts | grep devtmpfs
    udev /dev devtmpfs rw,nosuid,relatime,size=8149048k,nr_inodes=2037262,mode=755 0 0
    udev /proc/kcore devtmpfs rw,nosuid,relatime,size=8149048k,nr_inodes=2037262,mode=755 0 0
    udev /proc/timer_list devtmpfs rw,nosuid,relatime,size=8149048k,nr_inodes=2037262,mode=755 0 0
    udev /proc/timer_stats devtmpfs rw,nosuid,relatime,size=8149048k,nr_inodes=2037262,mode=755 0 0
    udev /proc/sched_debug devtmpfs rw,nosuid,relatime,size=8149048k,nr_inodes=2037262,mode=755 0 0

    $ docker run --rm -ti -v /dev:/dev --privileged ubuntu:14.04 cat /proc/mounts | grep devtmpfs
    udev /dev devtmpfs rw,nosuid,relatime,size=8149048k,nr_inodes=2037262,mode=755 0 0

Note that `{{ ... }}` is to escape Python's string `format` operation.
The shell sees `{ ... }`.

drothlis commented:
> This does make me a little nervous.  I'm thinking about the case where you have `/dev` mounted in from the host.  Maybe it's fine.  Also I don't think we would want `stbt-docker` to fail if `/dev/raw1394` existed or `/dev` was for some reason unwritable.

I've added a commit to address this, and I've rebased onto master to resolve conflicts.
